crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -0,0 +1,134 @@
+use super::Primitive2d;
+use crate::Vec2;
+
+/// A normalized vector pointing in a direction in 2D space
+#[derive(Clone, Copy, Debug)]
+pub struct Direction2d(Vec2);
+
+impl From<Vec2> for Direction2d {
+    fn from(value: Vec2) -> Self {
+        Self(value.normalize())
+    }
+}
+
+impl Direction2d {
+    /// Create a direction from a [Vec2] that is already normalized
+    pub fn from_normalized(value: Vec2) -> Self {
+        Self(value)
+    }
+}
+
+impl std::ops::Deref for Direction2d {
+    type Target = Vec2;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// An infinite half-line pointing in a direction in 2D space
+#[derive(Clone, Copy, Debug)]
+pub struct Ray2d(pub Direction2d);
+
+/// A circle primitive
+#[derive(Clone, Copy, Debug)]
+pub struct Circle {
+    /// The radius of the circle
+    pub radius: f32,
+}
+impl Primitive2d for Circle {}
+
+/// An unbounded plane in 2D space
```

Comment:
Probably worth adding another sentence to help people visualize that this is just a line.

Reply:
It's not. I was confused initially. Instead, it's a separating surface that passes through the origin, defined by its normal.

It's pretty weird to think about these in 2D, so we should be extra clear on them (if we want to have it).

Reply:
Oh yes of course, a "plane" in 2D is actually a line!

Reply:
It would show up as a line, just a weird one that has some concept of a front and back side. If you combine a few planes you can use it for culling for example. There's also things like half space colliders that are defined purely by a plane, everything facing away from the normal would be considered solid. Both aren't nearly as useful in 2D as 3D however. I just copied it from RFC to keep it consistent with 3D

Reply:
Yes, makes sense. I was just thinking about 3D the first time I reviewed. 🙃